platform/ios/Integration Tests/Snapshotter Tests/MGLMapSnapshotterTest.m

```diff
@@ -0,0 +1,239 @@
+        dispatch_group_wait(group, DISPATCH_TIME_FOREVER);
+
+        snapshotter = nil;
+
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            [expectation fulfill];
+        });
+    });
+
+    [self waitForExpectations:@[expectation] timeout:2.0];
+}
+
+- (void)testMultipleSnapshottersFromBackgroundQueue {
+    if (!validAccessToken()) {
+        return;
+    }
+
+    // Crashes with only 1 snapshot
+    CGSize size = self.mapView.bounds.size;
+    CLLocationCoordinate2D coord = CLLocationCoordinate2DMake(30.0, 30.0);
+
+    XCTestExpectation *expectation = [self expectationWithDescription:@"snapshots"];
+    expectation.expectedFulfillmentCount = 1;
+    expectation.assertForOverFulfill = YES;
+
+    __weak __typeof__(self) weakself = self;
+
+    dispatch_queue_attr_t attr = dispatch_queue_attr_make_with_qos_class(DISPATCH_QUEUE_SERIAL, QOS_CLASS_USER_INITIATED, QOS_MIN_RELATIVE_PRIORITY); // also for concurrent
+    dispatch_queue_t backgroundQueue = dispatch_queue_create(__PRETTY_FUNCTION__, attr);
+
+
+    // Use dispatch_group to keep the backgroundQueue block around (and
+    // so also the MGLMapSnapshotter
+    dispatch_group_t group = dispatch_group_create();
+    dispatch_group_enter(group);
+
+
+    dispatch_async(backgroundQueue, ^{
+
+        MGLMapSnapshotter *snapshotter = snapshotterWithCoordinates(coord, size);
+        XCTAssertNotNil(snapshotter);
+
+        MGLMapSnapshotCompletionHandler completion = ^(MGLMapSnapshot * _Nullable snapshot, NSError * _Nullable error) {
+
+            // This should be the main queue
+            __typeof__(self) strongself = weakself;
+
+            MGLTestAssertNotNil(strongself, strongself);
+
+            MGLTestAssertNotNil(strongself, snapshot);
+            MGLTestAssertNotNil(strongself, snapshot.image);
+            MGLTestAssertNil(strongself, error, @"Snapshot should not error with: %@", error);
+
+            // Change this back to XCTAttachmentLifetimeDeleteOnSuccess when we're sure this
+            // test is passing.
+            XCTAttachment *attachment = [XCTAttachment attachmentWithImage:snapshot.image];
+            attachment.lifetime = XCTAttachmentLifetimeKeepAlways;
+            [strongself addAttachment:attachment];
+
+            dispatch_group_leave(group);
+        };
+
+        // untested
+        @try {
+            [snapshotter startWithCompletionHandler:completion];
+            MGLTestFailWithSelf(weakself);
+        }
+        @catch (NSException *exception) {
+            MGLTestAssert(weakself, exception.name == NSInvalidArgumentException);
+            dispatch_group_leave(group);
+        }
+
+        // Wait for the snapshot to complete
+        dispatch_group_wait(group, DISPATCH_TIME_FOREVER);
+
+        snapshotter = nil;
+
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            [expectation fulfill];
+        });
+    });
+
+    [self waitForExpectations:@[expectation] timeout:10.0];
+}
+
+- (void)testMultipleSnapshotters {
```

Comment:
I think it's worth temporarily disabling the test so that we don't get false positives during CI. Let's just rename the failing methods something like `- (void)disabled_test....`

When we merge to master, we can mark this as a pending test (which uses a environment variable to decide whether to run or not)

Reply:
I don't know the details of how this runs on CI, but I agree we should disable tests that may fail intermittently. Could you disable the tests that you think need disabling?